### PR TITLE
Implemented input validation for the detector description field.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
Signed-off-by: AWSHurneyt <hurneyt@amazon.com>

### Description
Implemented input validation for the detector description field.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by getsaurabh02 — APPROVED** [review]

File: public/utils/validation.ts [hunk 1]
```diff
@@ -29,7 +29,34 @@ export function getNameErrorMessage(
...
+      upper and lowercase letters, numbers 0-9, hyphens, spaces, and underscores.`;
+  }
+}
+
+export const MIN_DESCRIPTION_CHARACTERS = 0;
+export const MAX_DESCRIPTION_CHARACTERS = 500;
+
+// This regex pattern support MIN to MAX character limit, capital and lowercase letters,
+// numbers 0-9, hyphens, periods, spaces, and underscores.
+export const DESCRIPTION_REGEX = new RegExp(/^[a-zA-Z0-9 _.,-]{0,500}$/);
```
Comment (bowenlan-amzn) [inline public/utils/validation.ts:41]: Do we need to export this or only used in below function

**Comment by lezzago** [comment]
Can we add unit tests for this?

File: public/pages/CreateDetector/components/DefineDetector/components/DetectorDetails/DetectorBasicDetailsForm.tsx [hunk 2]
```diff
@@ -69,12 +94,17 @@ export default class DetectorBasicDetailsForm extends Component<
...
+          isInvalid={descriptionFieldTouched && descriptionIsInvalid}
+          error={getNameErrorMessage(detectorName, nameIsInvalid, nameFieldTouched)}
+        >
           <EuiTextArea
             placeholder={'Enter a description for the detector.'}
             compressed={true}
             value={detectorDescription}
-            onChange={onDetectorInputDescriptionChange}
+            onBlure={this.onDescriptionBlur}
+            onChange={this.onDescriptionChange}
```
Comment (bowenlan-amzn) [inline public/pages/CreateDetector/components/DefineDetector/components/DetectorDetails/DetectorBasicDetailsForm.tsx:107]: I think you also want `error={getDescriptionErrorMessage(...)}` here

Comment (AWSHurneyt) [reply public/pages/CreateDetector/components/DefineDetector/components/DetectorDetails/DetectorBasicDetailsForm.tsx:111]: We call that here https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/84/files#diff-ce777ba621c03240471af2cfaa5a09eb86107b8ed94fd28a033b6913b2ef3515R100

**Comment by AWSHurneyt** [comment]
> Can we add unit tests for this?

We have plans to add unit tests, but for now, manual testing suggests the validation is working as expected.

File: public/pages/CreateDetector/components/DefineDetector/components/DetectorDetails/DetectorBasicDetailsForm.tsx [hunk 3]
```diff
@@ -45,22 +54,38 @@ export default class DetectorBasicDetailsForm extends Component<
...
+      descriptionIsInvalid: !validateDescription(event.target.value),
+    });
+  };
+
+  onDescriptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.props.onDetectorInputDescriptionChange(event.target.value.trimStart());
+  };
+
   render() {
     const { detectorName, detectorDescription, onDetectorInputDescriptionChange } = this.props;
```
Comment (amsiglan) [inline public/pages/CreateDetector/components/DefineDetector/components/DetectorDetails/DetectorBasicDetailsForm.tsx:69]: onDetectorInputDescriptionChange is no longer being used

**Review by amsiglan — APPROVED** [review]

**Pull request merged**